### ISSUE TITLE
Improvements to plane stress

### ISSUE THREAD

**Issue by bwspenc** [idaholab/moose#14484]
## Reason
The plane stress capability in TensorMechanics has not been fully developed.  It needs a couple more features to be fully functional and more usable.

## Design
The following features should be added to the plane stress system
- [ ] Add the ability to set up plane stress using the TensorMechanics/Master action
- [ ] Account for the thickness change in finite deformation scenarios to account for necking in the out of plane direction
- [ ] Provide support for defining the thickness in 2D planar models. The thickness should be handled at a low level in the framework so that it is taken into account for all objects that do volume integrals.

## Impact
Adding the ability to set it up with the Action will not change any existing functionality. This is also the case for adding thickness support. The modification for finite strain will affect current solutions, which don't take the thickness change into account.

**Comment by bwspenc** [comment]
FYI @hugary1995 and @hchen139. I am close to having a PR ready to address the first 2 of these issues.

**Comment by tophmatthews** [comment]
> * Account for the thickness change in finite deformation scenarios to account for necking in the out of plane direction

Does this include the ability to accumulate axial strain like we talked about yesterday?

**Comment by bwspenc** [comment]
@tophmatthews No, this is unrelated to accumulating the integrated axial strain over multiple layers. This just deals with the behavior within a single layer, and is applicable only to plane stress, which isn't typically used in fuel analysis.
